diff --git a/dist/jsx-directive-w2gUTWdx.js b/dist/jsx-directive-w2gUTWdx.js
index 637b42ba829bb5f90d4de93cc8ed1d562694b49f..0f94008f8487917273b3efc38ed784746ceb3f67 100644
--- a/dist/jsx-directive-w2gUTWdx.js
+++ b/dist/jsx-directive-w2gUTWdx.js
@@ -1,6 +1,4 @@
-import { addCode } from "./common-BTTMZY21.js";
 import { allCodeFeatures, createPlugin } from "ts-macro";
-import { createFilter } from "@vue-macros/common";
 import { isHTMLTag, isSVGTag } from "@vue/shared";
 
 //#region src/jsx-directive/common.ts
@@ -143,7 +141,7 @@ function transformCtx(node, root, index, options) {
 		const prefixModel = `${prefix}model`;
 		if (name.startsWith(prefixModel)) {
 			const dir = getDirectiveArgs(prop, options);
-			if (dir.valueCode) props += `${dir.isDynamic ? "[" : `'`}${dir.argument || dir.argumentCode?.[0] || "modelValue"}${dir.isDynamic ? "]" : `'`}: ${dir.valueCode[0]},`;
+			if (dir.valueCode && !dir.isDynamic) props += `'${dir.argument || dir.argumentCode?.[0].slice(1, -1) || 'modelValue'}': ${dir.valueCode[0]},`
 			continue;
 		} else if (name.includes("_")) name = name.split("_")[0];
 		else if (prefix && name.startsWith(prefix)) continue;
@@ -167,6 +165,10 @@ function transformCtx(node, root, index, options) {
 	else addCode(codes, result);
 	return ctxName;
 }
+function addCode(codes, ...args) {
+	const index = codes.findIndex((code) => code.includes("__VLS_setup = (async () => {"));
+	codes.splice(index === -1 ? codes.length : index + 1, 0, ...args);
+}
 function getRefValue(expression, options) {
 	const { ts, ast } = options;
 	if (ts.isIdentifier(expression)) return expression.getText(ast);
@@ -737,11 +739,10 @@ function getTagName(node, options) {
 //#region src/jsx-directive.ts
 const plugin = createPlugin(({ ts, vueCompilerOptions }, options = vueCompilerOptions?.vueMacros?.jsxDirective === true ? {} : vueCompilerOptions?.vueMacros?.jsxDirective ?? {}) => {
 	if (!options) return [];
-	const filter = createFilter(options);
 	return {
 		name: "vue-macros-jsx-directive",
-		resolveVirtualCode({ filePath, ast, codes, lang }) {
-			if (!filter(filePath) || !["jsx", "tsx"].includes(lang)) return;
+		resolveVirtualCode({ ast, codes, lang }) {
+			if (!["jsx", "tsx"].includes(lang)) return;
 			transformJsxDirective({
 				codes,
 				ast,
diff --git a/dist/jsx-directive.js b/dist/jsx-directive.js
index accace0c32b6a1cf5ceb297c8fbf07f22f438a23..5023f21444592741c43a4950f08fe959a452da4c 100644
--- a/dist/jsx-directive.js
+++ b/dist/jsx-directive.js
@@ -1,4 +1,3 @@
-import "./common-BTTMZY21.js";
 import { jsx_directive_default, plugin } from "./jsx-directive-w2gUTWdx.js";
 
 export { jsx_directive_default as default, plugin as "module.exports" };
\ No newline at end of file
diff --git a/dist/jsx-ref-DZlb-F7c.js b/dist/jsx-ref-DZlb-F7c.js
index ffe0b9f6eab21f435b60e4eeb32e0931d629a1bd..a874bc6b4b38e64ee4fa7ea061070498408301cf 100644
--- a/dist/jsx-ref-DZlb-F7c.js
+++ b/dist/jsx-ref-DZlb-F7c.js
@@ -1,5 +1,4 @@
 import { createPlugin } from "ts-macro";
-import { createFilter } from "@vue-macros/common";
 
 //#region src/jsx-ref.ts
 function transformRef({ nodes, codes, ts }) {
@@ -27,12 +26,11 @@ function getRefNodes(ts, sourceFile, alias) {
 }
 const plugin = createPlugin(({ ts, vueCompilerOptions }, options = vueCompilerOptions?.vueMacros?.jsxRef === true ? {} : vueCompilerOptions?.vueMacros?.jsxRef ?? {}) => {
 	if (!options) return [];
-	const filter = createFilter(options);
 	const alias = options.alias || ["useRef"];
 	return {
 		name: "vue-macros-jsx-ref",
-		resolveVirtualCode({ filePath, ast, codes, source, lang }) {
-			if (!filter(filePath) || !["jsx", "tsx"].includes(lang)) return;
+		resolveVirtualCode({ ast, codes, source, lang }) {
+			if (!["jsx", "tsx"].includes(lang)) return;
 			const nodes = getRefNodes(ts, ast, alias);
 			if (nodes.length) transformRef({
 				nodes,
