diff --git a/dist/jsx-directive-BhBU9QBv.js b/dist/jsx-directive-BhBU9QBv.js
index 92111393988d1144f4ac772d7a9930acf3e10da9..2eeeb2f26259c9b26417235428019dea3dff5cc8 100644
--- a/dist/jsx-directive-BhBU9QBv.js
+++ b/dist/jsx-directive-BhBU9QBv.js
@@ -1,6 +1,4 @@
-import { addCode } from "./common-BTTMZY21.js";
 import { allCodeFeatures, createPlugin } from "ts-macro";
-import { createFilter } from "@vue-macros/common";
 import { isHTMLTag, isSVGTag } from "@vue/shared";
 
 //#region src/jsx-directive/common.ts
@@ -143,7 +141,7 @@ function transformCtx(node, root, index, options) {
 		const prefixModel = `${prefix}model`;
 		if (name.startsWith(prefixModel)) {
 			const dir = getDirectiveArgs(prop, options);
-			if (dir.valueCode) props += `${dir.isDynamic ? "[" : `'`}${dir.argument || "modelValue"}${dir.isDynamic ? "]" : `'`}: ${dir.valueCode[0]},`;
+			if (dir.valueCode) props += `${dir.isDynamic ? "[" : `'`}${dir.argument || dir.argumentCode?.[0] || "modelValue"}${dir.isDynamic ? "]" : `'`}: ${dir.valueCode[0]},`;
 			continue;
 		} else if (name.includes("_")) name = name.split("_")[0];
 		else if (prefix && name.startsWith(prefix)) continue;
@@ -167,6 +165,10 @@ function transformCtx(node, root, index, options) {
 	else addCode(codes, result);
 	return ctxName;
 }
+function addCode(codes, ...args) {
+	const index = codes.findIndex((code) => code.includes("__VLS_setup = (async () => {"));
+	codes.splice(index === -1 ? codes.length : index + 1, 0, ...args);
+}
 function getRefValue(expression, options) {
 	const { ts, ast } = options;
 	if (ts.isIdentifier(expression)) return expression.getText(ast);
@@ -403,7 +405,7 @@ function transform(nodes, ctxMap, options) {
 				[modify, start + name.length + argument.length + (isDynamic ? 2 : 0) + 2 + (index ? modifiers.slice(0, index).join("").length + index : 0)],
 				modify ? `${modify.includes("-") ? `'` : ""}: true, ` : `'`
 			]), "} satisfies ", isDynamic ? "Record<string, boolean>" : modifierProps, "}");
-			emitsResult.push(`${isDynamic ? "[`" : `'`}onUpdate:${isDynamic ? "${" : ""}${argument || modelValue}${isDynamic ? "}`]" : `'`}: () => {}, `);
+			emitsResult.push(`${isDynamic ? "[`" : `'`}onUpdate:${isDynamic ? "${" : ""}${argument || argumentCode?.[0] || modelValue}${isDynamic ? "}`]" : `'`}: () => {}, `);
 		}
 	}
 	if (firstNamespacedNode) {
@@ -540,7 +542,7 @@ function transformVSlot(nodeMap, ctxMap, options) {
 				const node$1 = isSlotTemplate && ts.isJsxElement(child) ? child.children : child;
 				return isSlotTemplate && ts.isJsxSelfClosingElement(child) ? "" : [ast.text.slice(node$1.pos, node$1.end), node$1.pos];
 			}), "</>,");
-			if (vForAttribute) result.push("})),");
+			if (vForAttribute) result.push("})) as any,");
 			if (vIfAttribute && vIfAttributeName) {
 				if ([`${prefix}if`, `${prefix}else-if`].includes(vIfAttributeName)) {
 					const nextIndex = index + (attributes[index + 1]?.[0] ? 1 : 2);
@@ -736,11 +738,10 @@ function getTagName(node, options) {
 //#region src/jsx-directive.ts
 const plugin = createPlugin(({ ts, vueCompilerOptions }, options = vueCompilerOptions?.vueMacros?.jsxDirective === true ? {} : vueCompilerOptions?.vueMacros?.jsxDirective ?? {}) => {
 	if (!options) return [];
-	const filter = createFilter(options);
 	return {
 		name: "vue-macros-jsx-directive",
-		resolveVirtualCode({ filePath, ast, codes, lang }) {
-			if (!filter(filePath) || !["jsx", "tsx"].includes(lang)) return;
+		resolveVirtualCode({ ast, codes, lang }) {
+			if (!["jsx", "tsx"].includes(lang)) return;
 			transformJsxDirective({
 				codes,
 				ast,
diff --git a/dist/jsx-directive.js b/dist/jsx-directive.js
index e5626b7e24c1c565aa0811e6e9d2dadcba6b7f10..2fe889a0c7a154b961a46a9065796faa7aaffe14 100644
--- a/dist/jsx-directive.js
+++ b/dist/jsx-directive.js
@@ -1,4 +1,3 @@
-import "./common-BTTMZY21.js";
 import { jsx_directive_default, plugin } from "./jsx-directive-BhBU9QBv.js";
 
 export { jsx_directive_default as default, plugin as "module.exports" };
\ No newline at end of file
diff --git a/dist/jsx-ref-DZlb-F7c.js b/dist/jsx-ref-DZlb-F7c.js
index ffe0b9f6eab21f435b60e4eeb32e0931d629a1bd..a874bc6b4b38e64ee4fa7ea061070498408301cf 100644
--- a/dist/jsx-ref-DZlb-F7c.js
+++ b/dist/jsx-ref-DZlb-F7c.js
@@ -1,5 +1,4 @@
 import { createPlugin } from "ts-macro";
-import { createFilter } from "@vue-macros/common";
 
 //#region src/jsx-ref.ts
 function transformRef({ nodes, codes, ts }) {
@@ -27,12 +26,11 @@ function getRefNodes(ts, sourceFile, alias) {
 }
 const plugin = createPlugin(({ ts, vueCompilerOptions }, options = vueCompilerOptions?.vueMacros?.jsxRef === true ? {} : vueCompilerOptions?.vueMacros?.jsxRef ?? {}) => {
 	if (!options) return [];
-	const filter = createFilter(options);
 	const alias = options.alias || ["useRef"];
 	return {
 		name: "vue-macros-jsx-ref",
-		resolveVirtualCode({ filePath, ast, codes, source, lang }) {
-			if (!filter(filePath) || !["jsx", "tsx"].includes(lang)) return;
+		resolveVirtualCode({ ast, codes, source, lang }) {
+			if (!["jsx", "tsx"].includes(lang)) return;
 			const nodes = getRefNodes(ts, ast, alias);
 			if (nodes.length) transformRef({
 				nodes,
